roles/download-site/tasks/main.yml

```diff
@@ -27,18 +27,18 @@
 # KNOWN ISSUES:
 # --
 
-- name: Download copy of database from sites
-  shell: "drush @{{ server_alias }}.{{ site_prefix }}_{{ inventory_hostname }} sql-dump --structure-tables-key=common > /tmp/{{ inventory_hostname }}/dbdump.sql"
-  when: sites_drush_prefix == 'default'
-
 - name: Set sites drush alias
   set_fact:
-    sites_drush_alias: "{{ server_alias }}.{{ sites_drush_prefix }}{{ inventory_hostname }}"
-  when: sites_drush_prefix != 'default'
+    sites_drush_alias: '{% if sites_drush_prefix == "default" %}{{ server_alias }}.{{ site_prefix }}_{{ inventory_hostname }}{% else %}{{ server_alias }}.{{ sites_drush_prefix }}{{ inventory_hostname }}{% endif %}'
 
-- name: Download copy of database from sites when using a custom drush prefix
+- name: Put Sites 1.x site in maintenance mode for final pre-launch migration
```

Comment:
Shouldn't we put the site into maintenance mode after we copy it? Does maintenance mode effect any of the Drupal internals? I am assuming it does not as you have likely tested this already.

Reply:
I have not tested it with the many permutations of manipulations that we are doing to the database in these migration scripts; however, there is nothing in my experience as a Drupaller that indicates to me that it would do any harm to put it in maintenance mode before creating the SQL dump. It does avoid the (admittedly unlikely) scenario where something could be changed during the time when the DB dump is being created. That to me seems more potentially damaging than putting it in maintenance mode first.

Reply:
That's a fair point. I am more raising the question so that it is thought about rather than assumed.

Reply:
Aye, and thanks for questioning the assumption. Much like with the `image_allow_insecure_derivatives` conversation, it provides a reference point when Future Us look back and wonder WTF Past Us were thinking.

Reply:
So, this should have code comments, as should the `image_allow_insecure_derivatives` code.

Note to Now Us and Future Us - when there are misgivings about an approach, the misgivings should be documented in comments.